codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/BuilderGenerator.kt

```diff
@@ -217,17 +223,24 @@ class BuilderGenerator(
         docs("To override the contents of this collection use [`${member.setterName()}`](Self::${member.setterName()}).")
         rust("///")
         documentShape(member, model, autoSuppressMissingDocs = false)
+        val k = when (coreType.key) {
+            is RustType.String,
+            is RustType.Box -> "impl Into<${coreType.key.render(true)}>"
+            else -> "${coreType.key.render(true)}"
+        }
+        val v = when (coreType.member) {
+            is RustType.String,
+            is RustType.Box -> "impl Into<${coreType.member.render(true)}>"
+            else -> "${coreType.member.render(true)}"
+        }
```

Comment:
lets extract this snippet into a function

Reply:
I extracted the snippet but I feel like I did it in a very caveman manner and I'm open to suggestions if you think it's gross. I wasn't sure where they belonged and I also struggled to come up with informational names.